.github/workflows/ci.yml

```diff
@@ -0,0 +1,30 @@
+name: CI
+on: push
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Setup node
+        uses: actions/setup-node@v2
+
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          # npm cache files are stored in `~/.npm` on Linux/macOS
+          path: ~/.npm
+          key: ci-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ci-${{ env.cache-name }}-
+            ci-
+
+      - name: Install node modules
+        run: npm install
+
+      - name: Run tests
+        run: npm test
```

Comment:
Then we have all the steps which are a part of this job. They have a `name` which is presented in the [GitHub Actions UI](https://github.com/kmaaallen/health-log-app/pull/2/checks?check_run_id=3101758109) and either a `uses` or `run` value.

`uses` specifies an existing action which can be pulled from a repo like [`actions/checkout`](https://github.com/actions/checkout) and `run` takes a command which will be run directly like `npm test`.

The `Cache node modules` step provides some extra configuration which is supported by the [`actions/cache`](https://github.com/actions/cache) action. It sets an environment variable of the `cache-name` (mostly so we don't have to update it in multiple places if we decide to change it), and tells the cache action that we want to:
- cache the contents of the `~/.npm` directory (where node caches its modules before moving them into `./node_modules`)
- use a cache key of `ci-cache-node-modules-` with a hash of the `package-lock.json` file (the hash will be the same each time if the file hasn't changed, so it can just use exactly the same set of files as before)
- check for two other keys if we can't find an exact match - so if the `package-lock.json` file has changed, we get the last one that starts with `ci-cache-node-modules-` and if the `cache-name` value has changed we load anything with `ci-`

The cache step will also add a 'post' step which runs when the whole job has completed - this is where we save the cache (you can see this in the run for this action: https://github.com/kmaaallen/health-log-app/pull/2/checks?check_run_id=3101758109).